zipkin-lens/src/components/DiscoverPage/DiscoverPageContent.test.jsx

```diff
@@ -0,0 +1,221 @@
+    const wrapper = ({ children }) => {
+      return <Router history={history}>{children}</Router>;
+    };
+    history.push({
+      pathname: '/zipkin/',
+      search: '?limit=300',
+    });
+
+    const { result } = renderHook(() => useQueryParams(['key3']), { wrapper });
+
+    act(() => {
+      result.current.setQueryParams(
+        [
+          { key: 'serviceName', value: 'serviceA' },
+          { key: 'spanName', value: 'spanB' },
+          { key: 'remoteServiceName', value: 'remoteServiceNameC' },
+          // Durations will NOT converted to microsecond values.
+          { key: 'minDuration', value: '10us' },
+          { key: 'maxDuration', value: '100ms' },
+          // AnnotationQuery
+          { key: 'tagQuery', value: 'key1=value1 and key2' },
+          { key: 'key3', value: 'value3' },
+        ],
+        {
+          type: 'fixed',
+          endTime: moment(1588558961791),
+          value: '2h',
+        },
+        10,
+      );
+    });
+    expect(history.location.search).toBe(
+      '?serviceName=serviceA&spanName=spanB&remoteServiceName=remoteServiceNameC&minDuration=10us&maxDuration=100ms&annotationQuery=key1%3Dvalue1+and+key2+and+key3%3Dvalue3&lookback=2h&endTs=1588558961791&limit=10',
+    );
+  });
+});
+
+it('parseDuration', () => {
+  [
+    { in: '35', out: 35 },
+    { in: '35us', out: 35 },
+    { in: '35ms', out: 35 * 1000 },
+    { in: '35s', out: 35 * 1000 * 1000 },
+  ].forEach((e) => {
+    expect(parseDuration(e.in)).toBe(e.out);
+  });
+});
+
+describe('buildApiQuery', () => {
+  it('should build API Query', () => {
+    const params = buildApiQuery(
+      [
+        { key: 'serviceName', value: 'serviceA' },
+        { key: 'spanName', value: 'spanB' },
+        { key: 'remoteServiceName', value: 'remoteServiceNameC' },
+        // Durations will converted to microsecond values.
+        { key: 'minDuration', value: '10us' },
+        { key: 'maxDuration', value: '100ms' },
+        // AnnotationQuery
+        { key: 'tagQuery', value: 'key1=value1 and key2' },
+        { key: 'key3', value: 'value3' },
+      ],
+      {
+        type: 'fixed',
+        endTime: moment(1588558961791),
+        value: '2h',
+      },
+      30,
+      ['key3'],
+    );
+    expect(params.serviceName).toBe('serviceA');
+    expect(params.spanName).toBe('spanB');
+    expect(params.remoteServiceName).toBe('remoteServiceNameC');
+    expect(params.minDuration).toBe('10');
+    expect(params.maxDuration).toBe('100000');
+    expect(params.annotationQuery).toBe('key1=value1 and key2 and key3=value3');
+    expect(params.lookback).toBe('7200000');
+    expect(params.endTs).toBe('1588558961791');
+    expect(params.limit).toBe('30');
+  });
+});
+
+describe('<DiscoverPageContent />', () => {
+  it('should initialize the lookback using config.json', () => {
+    const { getAllByText } = render(<DiscoverPageContent />, {
+      uiConfig: {
+        defaultLookback: 60 * 1000 * 5, // 5m
```

Comment:
More specifically, we should have a test where this is set to an arbitrary milliseconds like 1234567